src/MagnumPlugins/AssimpImporter/AssimpImporter.cpp

```diff
@@ -1118,6 +1135,235 @@ Containers::Optional<ImageData2D> AssimpImporter::doImage2D(const UnsignedInt id
+            const aiNodeAnim* channel = animation->mChannels[c];
+
+            /** @todo handle alignment once we do more than just four-byte types */
+            dataSize += channel->mNumPositionKeys*(sizeof(Float) + sizeof(Vector3)) +
+                channel->mNumRotationKeys*(sizeof(Float) + sizeof(Quaternion)) +
+                channel->mNumScalingKeys*(sizeof(Float) + sizeof(Vector3));
+
+            trackCount += (channel->mNumPositionKeys > 0 ? 1 : 0) +
+                (channel->mNumRotationKeys > 0 ? 1 : 0) +
+                (channel->mNumScalingKeys  > 0 ? 1 : 0);
+        }
+    }
+
+    /* Populate the data array */
+    Containers::Array<char> data{dataSize};
+
+    /* Import all tracks */
+    bool hadToRenormalize = false;
+    std::size_t dataOffset = 0;
+    std::size_t trackId = 0;
+    Containers::Array<Trade::AnimationTrackData> tracks{trackCount};
+    for(std::size_t a = animationBegin; a != animationEnd; ++a) {
+        const aiAnimation* animation = _f->scene->mAnimations[a];
+        for(std::size_t c = 0; c != animation->mNumChannels; ++c) {
+            const aiNodeAnim* channel = animation->mChannels[c];
+            const UnsignedInt target = doObject3DForName(channel->mNodeName.C_Str());
+            if(target == -1) {
+                Error{} << "Trade::AssimpImporter::animation(): unknown target node " << channel->mNodeName.C_Str();
+                return Containers::NullOpt;
+            }
+
+            /* Assimp only supports linear interpolation. For glTF splines
+               it simply uses the spline control points. */
+            /** @todo Before https://github.com/assimp/assimp/commit/e3083c21f0a7beae6c37a2265b7919a02cbf83c4
+             *        Assimp incorrectly read the spline tangents as values.
+             *        Is there a way to detect this so we can work around it,
+             *        or at least emit a warning?
+             */
+            constexpr Animation::Interpolation interpolation = Animation::Interpolation::Linear;
+
+            /* Extrapolation modes */
+            const Animation::Extrapolation extrapolationBefore = extrapolationFor(channel->mPreState);
+            const Animation::Extrapolation extrapolationAfter = extrapolationFor(channel->mPostState);
+
+            /* Key times are given as ticks, convert to milliseconds. Default value of 25 is
+               taken from the assimp_view tool. */
+            double ticksPerSecond = animation->mTicksPerSecond != 0.0 ? animation->mTicksPerSecond : 25.0;
+
+            /* For glTF files mTicksPerSecond is completely useless before
+               https://github.com/assimp/assimp/commit/7a8b7ba88d6d84dde7fd43419ac2b022c9887856
+               but can be assumed to always be 1000. */
+            /** @todo Check if this is broken in other importers, too */
+            const bool isGltf = std::strcmp(_f->importerName, "glTF2 Importer") == 0;
+            if(isGltf) ticksPerSecond = 1000.0;
```

Comment:
~~For shits and giggles~~ To inform the user, I would print a warning pointing out to assimp's brokeness:

```cpp
if(isGltf && !Math::equals(animation->mTicksPerSecond, 1000.0)) {
    Warning{} << "Trade::AssimpImporter::animation():" << animation->mTicksPerSecond << "ticks per second is incorrect for glTF, patching to 1000.0";
    ticksPerSecond = 1000.0;
}
```

At first I thought this could be only a verbose warning (`if(flags() & ImporterFlag::Verbose)`) but since it seems to be fixed in master and is a pretty important fix, I think it makes sense to print it always.

Reply:
Done 🙇